python/ray/serve/kv_store.py

```diff
@@ -56,3 +60,143 @@ def delete(self, key):
         if not isinstance(key, str):
             raise TypeError("key must be a string, got: {}.".format(type(key)))
         return ray_kv._internal_kv_del(self._format_key(key))
+
+
+class RayExternalKVStore:
+    """Persistent version of RayInternalKVStore for cluster fault
+    tolerance. Writes to local disk in local / test mode, also external
+    storage (such as amazon S3) with provided path.
+
+    Supports string keys and bytes values, caller must handle serialization.
+    """
+
+    def __init__(
+            self,
+            namepsace: str,
+            bucket="",
+            s3_path="",
+            region_name="us-west-2",
+            aws_access_key_id=None,
+            aws_secret_access_key=None,
+            aws_session_token=None,
+            local_mode=True,
+    ):
+        self._namespace = namepsace
+        self._bucket = bucket
+        self._s3_path = s3_path
+        self._local_mode = local_mode
```

Comment:
Hmm the local_mode flag is kind of weird given that it uses a completely separate code path for every operation, can we just make these two separate stores? `LocalDiskKVStore` and `S3KVStore`